PR: Upload file fix

Daniela-git opened the pull request.

Daniela-git committed:
add the launch.json to make debug

Daniela-git committed:
spaces

Daniela-git committed:
fixed the conflit in .travis

Daniela-git committed:
Merge branch 'travis' into depurar

Daniela-git committed:
start the iframe test

Daniela-git committed:
the frame change its height

Daniela-git committed:
get the new iframe height

Daniela-git committed:
created getTitle and switchToFrame

Daniela-git committed:
chage to the frame and get its title

Daniela-git committed:
created the getFrameTitle to divide switchFrame

Daniela-git committed:
return to the main page

Daniela-git committed:
added the file url to the json

Daniela-git committed:
Merge branch 'depurar'

Daniela-git committed:
improvement in the test

Daniela-git committed:
spaces

Daniela-git committed:
spaces

Daniela-git committed:
spaces

Daniela-git committed:
Merge branch 'master' into upload_file

Daniela-git committed:
Merge branch 'master' of https://github.com/Daniela-git/protractor-workshop-2021

Daniela-git committed:
fixing conflits

Daniela-git committed:
quit the last describe

merge the last two describe in one

Daniela-git committed:
quit the last describe

merge the last two describe in one

Daniela-git committed:
fix little conflict

Daniela-git committed:
fix some conflits

Daniela-git committed:
Merge branch 'master' of https://github.com/Daniela-git/protractor-workshop-2021

Daniela-git committed:
chage in two names

Daniela-git committed:
Merge branch 'master' into upload_file

Daniela-git committed:
using resolve for the path

Daniela-git committed:
using resolve for the path

Daniela-git committed:
Merge branch 'upload_file' of https://github.com/Daniela-git/protractor-workshop-2021 into upload_file

Daniela-git committed:
fix conflit

germandavid85 commented on test/locators.spec.ts:
```diff
@@ -24,9 +23,19 @@ describe('fill form', () => {
+    });
+  });
+  it('then should have a photo', async () => {
+    const files: number = await personalInformationPage.getFiles();
```
let's use the hook functions (before, beforeEach, etc) to organize the actions executed by the test, and the `it`s only for the `expect`s

Daniela-git committed:
changes in the locators spec

put all the test part in the before

holgiosalos commented on src/page/personal-information.page.ts:
```diff
@@ -41,23 +44,35 @@ export class PersonalInformation {
     return await this.title.getText();
   }
+
+  public async getFiles(): Promise<number> {
+    const numberOfFiles: number = await browser.executeScript(
+      ' return arguments[0].files.length;',
+      this.profile
+    );
+    return numberOfFiles;
```
Is it possible to get the name of the files instead of the number? then in the assertion, you verify that the correct file was uploaded using the name instead of using the number of files

Daniela-git committed:
change getFile function

now we get the name of the file that was upload to check if it's the same we used
